### ISSUE TITLE
random_shuffle=True but not random 

### ISSUE THREAD

**Issue by bigtaotao** [NVIDIA/DALI#1253]
There is my code:
`import os
import time
import torch.utils.data
import nvidia.dali.ops as ops
import nvidia.dali.types as types
import torchvision.datasets as datasets
from nvidia.dali.pipeline import Pipeline
import torchvision.transforms as transforms
from nvidia.dali.plugin.pytorch import DALIClassificationIterator, DALIGenericIterator

dataset_dir = '/dev/shm/gan/data_test'
index_file = '/dev/shm/gan/index.txt'
output_file = '/dev/shm/gan/output.txt'

def getImagePathList(dataset_dir):
    imagePathList = []
    print("ok")
    for root, dirs, files in os.walk(dataset_dir):
        
        for file in files:
            imagePathList.append(os.path.join(root, file))
    new = []
    for each in imagePathList:
        new.append(os.path.relpath(each))
    print("got_list")
    return new

class HybridTrainPipe(Pipeline):
    def __init__(self, batch_size, num_threads, device_id, data_dir, crop, dali_cpu=False, local_rank=0, world_size=1):
        super(HybridTrainPipe, self).__init__(batch_size, num_threads, device_id, seed=12 + device_id)
        dali_device = "gpu"
       
        self.input = ops.FileReader(file_root=data_dir,file_list=index_file, shard_id=local_rank, num_shards=world_size, random_shuffle=True)
        self.decode = ops.ImageDecoder(device="mixed", output_type=types.RGB)
        self.res = ops.RandomResizedCrop(device="gpu", size=crop, random_area=[0.08, 1.25])
        self.cmnp = ops.CropMirrorNormalize(device="gpu",
                                            output_dtype=types.FLOAT,
                                            output_layout=types.NCHW,
                                            image_type=types.RGB,
                                            mean=[0.485 * 255, 0.456 * 255, 0.406 * 255],
                                            std=[0.229 * 255, 0.224 * 255, 0.225 * 255])
        self.resize = ops.Resize(device = "gpu",
                                 image_type = types.RGB,
                                 interp_type = types.INTERP_LINEAR,
                                  resize_x = 2.0,
                                  resize_y = 2.0)
        self.coin = ops.CoinFlip(probability=0.5)
        print('DALI "{0}" variant'.format(dali_device))

    def define_graph(self):
        rng = self.coin()
        self.jpegs, self.labels = self.input(name="Reader")
        images = self.decode(self.jpegs)
        output = self.resize(images)
        #images = self.res(images)
        #output = self.cmnp(images, mirror=rng)
        return [output, self.labels]

def get_imagenet_iter_dali(type, image_dir, batch_size, num_threads, device_id, num_gpus, crop, val_size=256,
                           world_size=1,
                           local_rank=0):
    if type == 'train':
        pip_train = HybridTrainPipe(batch_size=batch_size, num_threads=num_threads, device_id=local_rank,
                                    data_dir=image_dir,crop=crop
                                    )
        pip_train.build()
        dali_iter_train = DALIClassificationIterator(pip_train, size=pip_train.epoch_size("Reader") // world_size)
        return dali_iter_train

def make_index(image_list,path):
    image_dic = {}
    for each in image_list:
        image_dic[each] = each.split('/')[-1].split(".")[0]
    with open(path,"w") as f:
        for each in image_dic:
            f.writelines(each + " " + image_dic[each] + "\n")
    return image_dic

os.chdir(dataset_dir)
continue_flag = 0
if continue_flag == 1:
     with open(index_file,"r") as f:
            file = f.readlines()
            image_list = []
            for each in file:
                image_list.append(each.split(" ")[0])
else:
    image_list = getImagePathList(dataset_dir)

print(image_list)
image_dic = make_index(image_list,index_file)

train_loader = get_imagenet_iter_dali(type='train', image_dir=dataset_dir, batch_size=1,
                                                num_threads=2, crop=224, device_id=1, num_gpus=1)
print('start iterate')

for i, data in enumerate(train_loader):
    labels = data[0]["label"]

    print(labels.cpu())
`

and this is output , I ran it twice:

[1]:
ok
got_list
['all/72.jpeg', 'all/69.jpeg', 'all/66.jpeg', 'all/65.jpeg', 'all/64.jpeg', 'all/63.jpeg', 'all/62.png', 'all/60.jpeg', 'all/59.jpeg', 'all/58.jpeg', 'all/54.jpeg', 'all/53.jpeg', 'all/52.jpeg', 'all/51.png', 'all/50.jpeg', 'all/48.jpeg', 'all/45.jpeg', 'all/44.jpeg', 'all/43.jpeg', 'all/40.jpeg', 'all/39.jpeg', 'all/38.jpeg', 'all/36.jpeg', 'all/35.jpeg', 'all/34.jpeg', 'all/33.jpeg', 'all/32.jpeg', 'all/31.jpeg', 'all/29.jpeg', 'all/28.jpeg', 'all/27.jpeg', 'all/26.jpeg', 'all/25.png', 'all/24.jpeg', 'all/23.jpeg', 'all/22.jpeg', 'all/21.jpeg', 'all/19.png', 'all/18.jpeg', 'all/17.jpeg', 'all/16.jpeg', 'all/15.png', 'all/14.jpeg', 'all/12.jpeg', 'all/11.jpeg', 'all/10.jpeg', 'all/9.jpeg', 'all/8.jpeg', 'all/5.jpeg']
DALI "gpu" variant
start iterate
tensor([[15]], dtype=torch.int32)
tensor([[44]], dtype=torch.int32)
tensor([[29]], dtype=torch.int32)
tensor([[23]], dtype=torch.int32)
tensor([[51]], dtype=torch.int32)
tensor([[72]], dtype=torch.int32)
tensor([[44]], dtype=torch.int32)
tensor([[40]], dtype=torch.int32)
tensor([[51]], dtype=torch.int32)
tensor([[65]], dtype=torch.int32)
libpng warning: iCCP: known incorrect sRGB profile
tensor([[24]], dtype=torch.int32)
tensor([[62]], dtype=torch.int32)
tensor([[58]], dtype=torch.int32)
tensor([[65]], dtype=torch.int32)
tensor([[31]], dtype=torch.int32)
tensor([[59]], dtype=torch.int32)
tensor([[59]], dtype=torch.int32)
tensor([[31]], dtype=torch.int32)
tensor([[21]], dtype=torch.int32)
tensor([[22]], dtype=torch.int32)
tensor([[51]], dtype=torch.int32)
tensor([[18]], dtype=torch.int32)
tensor([[63]], dtype=torch.int32)
tensor([[52]], dtype=torch.int32)
tensor([[48]], dtype=torch.int32)
tensor([[34]], dtype=torch.int32)
tensor([[18]], dtype=torch.int32)
tensor([[27]], dtype=torch.int32)
tensor([[35]], dtype=torch.int32)
tensor([[54]], dtype=torch.int32)
tensor([[24]], dtype=torch.int32)
tensor([[29]], dtype=torch.int32)
tensor([[43]], dtype=torch.int32)
tensor([[53]], dtype=torch.int32)
tensor([[10]], dtype=torch.int32)
tensor([[63]], dtype=torch.int32)
tensor([[8]], dtype=torch.int32)
tensor([[50]], dtype=torch.int32)
tensor([[34]], dtype=torch.int32)
tensor([[27]], dtype=torch.int32)
tensor([[59]], dtype=torch.int32)
tensor([[45]], dtype=torch.int32)
tensor([[58]], dtype=torch.int32)
tensor([[15]], dtype=torch.int32)
tensor([[69]], dtype=torch.int32)
tensor([[65]], dtype=torch.int32)
tensor([[15]], dtype=torch.int32)
tensor([[14]], dtype=torch.int32)
tensor([[53]], dtype=torch.int32)
libpng warning: iCCP: known incorrect sRGB profile

[2]:
ok
got_list
['all/72.jpeg', 'all/69.jpeg', 'all/66.jpeg', 'all/65.jpeg', 'all/64.jpeg', 'all/63.jpeg', 'all/62.png', 'all/60.jpeg', 'all/59.jpeg', 'all/58.jpeg', 'all/54.jpeg', 'all/53.jpeg', 'all/52.jpeg', 'all/51.png', 'all/50.jpeg', 'all/48.jpeg', 'all/45.jpeg', 'all/44.jpeg', 'all/43.jpeg', 'all/40.jpeg', 'all/39.jpeg', 'all/38.jpeg', 'all/36.jpeg', 'all/35.jpeg', 'all/34.jpeg', 'all/33.jpeg', 'all/32.jpeg', 'all/31.jpeg', 'all/29.jpeg', 'all/28.jpeg', 'all/27.jpeg', 'all/26.jpeg', 'all/25.png', 'all/24.jpeg', 'all/23.jpeg', 'all/22.jpeg', 'all/21.jpeg', 'all/19.png', 'all/18.jpeg', 'all/17.jpeg', 'all/16.jpeg', 'all/15.png', 'all/14.jpeg', 'all/12.jpeg', 'all/11.jpeg', 'all/10.jpeg', 'all/9.jpeg', 'all/8.jpeg', 'all/5.jpeg']
DALI "gpu" variant
start iterate
tensor([[15]], dtype=torch.int32)
tensor([[44]], dtype=torch.int32)
tensor([[29]], dtype=torch.int32)
tensor([[23]], dtype=torch.int32)
tensor([[51]], dtype=torch.int32)
tensor([[72]], dtype=torch.int32)
tensor([[44]], dtype=torch.int32)
tensor([[40]], dtype=torch.int32)
tensor([[51]], dtype=torch.int32)
tensor([[65]], dtype=torch.int32)
libpng warning: iCCP: known incorrect sRGB profile
tensor([[24]], dtype=torch.int32)
tensor([[62]], dtype=torch.int32)
tensor([[58]], dtype=torch.int32)
tensor([[65]], dtype=torch.int32)
tensor([[31]], dtype=torch.int32)
tensor([[59]], dtype=torch.int32)
tensor([[59]], dtype=torch.int32)
tensor([[31]], dtype=torch.int32)
tensor([[21]], dtype=torch.int32)
tensor([[22]], dtype=torch.int32)
tensor([[51]], dtype=torch.int32)
tensor([[18]], dtype=torch.int32)
tensor([[63]], dtype=torch.int32)
tensor([[52]], dtype=torch.int32)
tensor([[48]], dtype=torch.int32)
tensor([[34]], dtype=torch.int32)
tensor([[18]], dtype=torch.int32)
tensor([[27]], dtype=torch.int32)
tensor([[35]], dtype=torch.int32)
tensor([[54]], dtype=torch.int32)
tensor([[24]], dtype=torch.int32)
tensor([[29]], dtype=torch.int32)
tensor([[43]], dtype=torch.int32)
tensor([[53]], dtype=torch.int32)
tensor([[10]], dtype=torch.int32)
tensor([[63]], dtype=torch.int32)
tensor([[8]], dtype=torch.int32)
tensor([[50]], dtype=torch.int32)
tensor([[34]], dtype=torch.int32)
tensor([[27]], dtype=torch.int32)
tensor([[59]], dtype=torch.int32)
tensor([[45]], dtype=torch.int32)
tensor([[58]], dtype=torch.int32)
tensor([[15]], dtype=torch.int32)
tensor([[69]], dtype=torch.int32)
tensor([[65]], dtype=torch.int32)
tensor([[15]], dtype=torch.int32)
tensor([[14]], dtype=torch.int32)
tensor([[53]], dtype=torch.int32)
libpng warning: iCCP: known incorrect sRGB profile

[1] and [2]is totally same,I have no idea what to do
Can you help me?I will be deeply grateful
